src/main/java/uk/ac/sanger/sccp/stan/service/releasefile/ReleaseFileService.java

```diff
@@ -1042,13 +1050,39 @@ static boolean opSupplants(Operation newOp, Operation savedOp) {
     }
 
     /**
-     * Loads the stain types for the given operations.
-     * @param lwOps map to operations
-     * @return a map from operation id to stain types
+     * Loads flags relevant to the specified labware.
+     * @param entries the release entries
      */
-    public Map<Integer, List<StainType>> loadStainTypes(Map<Integer, Operation> lwOps) {
-        Set<Integer> opIds = lwOps.values().stream().map(Operation::getId).collect(toSet());
-        return stainTypeRepo.loadOperationStainTypes(opIds);
+    public void loadFlags(Collection<ReleaseEntry> entries) {
+        UCMap<Labware> labware = new UCMap<>();
+        for (ReleaseEntry entry : entries) {
+            Labware lw = entry.getLabware();
+            labware.put(lw.getBarcode(), lw);
+        }
+        List<FlagDetail> flagDetails = flagLookupService.lookUpDetails(labware.values());
+        if (flagDetails.isEmpty()) {
+            return;
+        }
+        Map<Integer, String> flagMap = flagDetails.stream()
+                .collect(toMap(fd -> labware.get(fd.getBarcode()).getId(),
+                        fd -> describeFlags(fd.getFlags())));
+        for (ReleaseEntry entry : entries) {
+            entry.setFlagDescription(flagMap.getOrDefault(entry.getLabware().getId(), ""));
```

Comment:
Would you please explain this? Why is it setting to an empty string?

Reply:
The story says that when there are no corresponding flags, the value should be empty. Here, if there are no flags for the labware we're looking at, we set the field to be an empty string.